geode-assembly/build.gradle

```diff
@@ -338,33 +350,34 @@ task gfshDepsJar (type: Jar, dependsOn: ':geode-core:classes') {
 
 def docsDir = file("$buildDir/javadocs")
 task docs(type: Javadoc) {
-    options.addStringOption('Xdoclint:none', '-quiet')
-    options.links("https://lucene.apache.org/core/6_4_1/core/")
-    options.links("https://lucene.apache.org/core/6_4_1/queryparser/")
-    options.encoding='UTF-8'
-    source parent.subprojects*.javadoc*.source
-    classpath = files(parent.subprojects*.javadoc*.classpath)
-    title = "${productName} ${project.version}"
-    include 'org/apache/geode/**/'
-    exclude 'org/apache/geode/internal/**/'
-    exclude 'org/apache/geode/**/internal/**/'
-    exclude 'org/apache/geode/**/xml/**/'
-    exclude 'org/apache/geode/distributed/**/util/**/'
-    exclude 'org/apache/geode/test/**/'
-    destinationDir = docsDir
+  options.addStringOption('Xdoclint:none', '-quiet')
+  options.links("https://lucene.apache.org/core/6_4_1/core/")
+  options.links("https://lucene.apache.org/core/6_4_1/queryparser/")
+  options.links("https://docs.oracle.com/javase/8/docs/api/")
+  options.encoding = 'UTF-8'
+  title = "${productName} ${project.version}"
+  destinationDir = docsDir
+
+  configurations.getByName("javadocOnly").allDependencies.each() { proj ->
+    proj.dependencyProject.tasks.withType(Javadoc).findAll { it.enabled }.each { javadocTask ->
+      source += javadocTask.source
+      classpath += javadocTask.classpath
+      excludes += javadocTask.excludes
+      includes += javadocTask.includes
+    }
+  }
+
+  include 'org/apache/geode/**/'
 
+  doLast {
     parent.subprojects.each { project ->
-      dependsOn(project.tasks['javadoc'])
```

Comment:
It appears the dependency of docs -> [project].javadoc was removed. Is that intentional?